components/TableOfContents.tsx

```
@@ -53,26 +132,47 @@ const TableOfContents: React.FC<TableOfContentsProps> = ({ items }) => {
           >
-            <span className="text-sm leading-relaxed">{item.text}</span>
-          </button>
+            <button
+              onClick={() => handleClick(item.id)}
+              data-id={item.id}
+              className={`text-left w-full transition-all duration-200 block py-2 px-3 rounded-md hover:bg-primary-50 dark:hover:bg-primary-900/30 ${
+                activeId === item.id
+                  ? 'text-primary-600 dark:text-primary-400 font-medium bg-primary-50 dark:bg-primary-900/30 border-l-2 border-primary-600 dark:border-primary-400'
+                  : 'text-secondary-600 dark:text-secondary-400 hover:text-primary-600 dark:hover:text-primary-400'
+              }`}
+            >
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)